cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1362,7 +1362,9 @@ private boolean hasRebalanceAnnotation(KafkaRebalance kafkaRebalance) {
 
     private boolean shouldRenewRebalance(KafkaRebalance kafkaRebalance, KafkaRebalanceAnnotation rebalanceAnnotation) {
         // check if refresh annotation applied or request are made when CC is not active
-        if (rebalanceAnnotation == KafkaRebalanceAnnotation.refresh || kafkaRebalance.getStatus().getConditions().stream().anyMatch(condition -> "ConnectException".equals(condition.getReason()))) {
+        if (rebalanceAnnotation == KafkaRebalanceAnnotation.refresh
+                || kafkaRebalance.getStatus().getConditions().stream().anyMatch(condition -> "ConnectException".equals(condition.getReason()))
+                || kafkaRebalance.getStatus().getConditions().stream().anyMatch(condition -> "AnnotatedNoRouteToHostException".equals(condition.getReason()))) {
```

Comment:
Tbh I suggested to use the base class `NoRouteToHostException` as we catch the `ConnectException` and not the derived `AnnotatedConnectException`. They are part of the Java base and not Netty. @ShubhamRwt @im-konge can you try with using `NoRouteToHostException` if it works for you?